### PR TITLE
Added Bitrise / Fastlane doc

### PR DESCRIPTION
**Pull request by RikH81**
Added guide to Fastlane integration with Bitrise CI

### CODE REVIEW

**Comment by RikH81** [comment]
I signed the CLA 🚀

**Comment by RikH81** [comment]
Content partly originates in our Devcenter, where it was authored by our Tech writer team: I suppose that triggered this issue with author consent verification?

**Comment by janpio** [comment]
Nah, it was because you accepted my suggestion - and @googlebot is not smart enough to understand that I consented already by creating the suggestion here (the GitHub feature of suggestions is quite new after all) and I also have a CLA of course. Nothing to worry about - I know that all is good and can merge this later (after now giving it a proper read).

File: docs/best-practices/continuous-integration/bitrise.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+[Bitrise](https://www.bitrise.io) allows you to automatically build, test and deploy your iOS and Android apps.
+
+## How to get started?
+
+Using _fastlane_ for your Bitrise workflow is simple. Just [add the ](https://devcenter.bitrise.io/getting-started/manage-your-bitrise-workflow) [_fastlane_](https://www.bitrise.io/integrations/steps/fastlane) [step to your
+workflow](https://devcenter.bitrise.io/getting-started/manage-your-bitrise-workflow),
+after the `Git Clone` step (and any other dependency step).
+
+### _Have our Certificate and profile installer step in your workflow!_
+_You should also add/keep the `Certificate and profile installer` step in the workflow, to download your .p12 Certificates and Provisioning Profiles uploaded to [bitrise.io](https://www.bitrise.io) and to install them. **Even if you don't upload your files to** [**bitrise.io**](https://www.bitrise.io) **and instead you use a_ _fastlane_ _tool to manage your code signing files you should still keep this step in the workflow**. Read more about [iOS Code Signing using third party tools](https://devcenter.bitrise.io/ios/code-signing/#use-a-third-party-tool-to-manage-your-code-signing-files)._
```
Comment (janpio) [inline docs/best-practices/continuous-integration/bitrise.md]: This line doesn't render properly: https://github.com/fastlane/docs/blob/cf1cd9a565ff9d8ee66e719260bf28f7619029dd/docs/best-practices/continuous-integration/bitrise.md

**Comment by RikH81** [comment]
Awesome. Thanks!

File: docs/best-practices/continuous-integration/bitrise.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+[Bitrise](https://www.bitrise.io) allows you to automatically build, test and deploy your iOS and Android apps.
+
+## How to get started?
+
+Using _fastlane_ for your Bitrise workflow is simple. Just [add the ](https://devcenter.bitrise.io/getting-started/manage-your-bitrise-workflow) [_fastlane_](https://www.bitrise.io/integrations/steps/fastlane) [step to your
+workflow](https://devcenter.bitrise.io/getting-started/manage-your-bitrise-workflow),
+after the `Git Clone` step (and any other dependency step).
+
+### _Have our Certificate and profile installer step in your workflow!_
```
Comment (janpio) [inline docs/best-practices/continuous-integration/bitrise.md]: Can we get rid of the italic here? It makes the whole paragraph a lot harder to read.

File: docs/best-practices/continuous-integration/bitrise.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
+# Bitrise Integration
+
+[Bitrise](https://www.bitrise.io) allows you to automatically build, test and deploy your iOS and Android apps.
+
+## How to get started?
+
+Using _fastlane_ for your Bitrise workflow is simple. Just [add the ](https://devcenter.bitrise.io/getting-started/manage-your-bitrise-workflow) [_fastlane_](https://www.bitrise.io/integrations/steps/fastlane) [step to your
```
Comment (janpio) [inline docs/best-practices/continuous-integration/bitrise.md]: The linking going on here is pretty confusing. Can this be rewritten so these two links have a word in between somehow and a link is not repeated?

(also: additional space after `add the` that should be removed)

File: docs/best-practices/continuous-integration/bitrise.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+workflow](https://devcenter.bitrise.io/getting-started/manage-your-bitrise-workflow),
+after the `Git Clone` step (and any other dependency step).
+
+### _Have our Certificate and profile installer step in your workflow!_
+_You should also add/keep the `Certificate and profile installer` step in the workflow, to download your .p12 Certificates and Provisioning Profiles uploaded to [bitrise.io](https://www.bitrise.io) and to install them. **Even if you don't upload your files to** [**bitrise.io**](https://www.bitrise.io) **and instead you use a_ _fastlane_ _tool to manage your code signing files you should still keep this step in the workflow**. Read more about [iOS Code Signing using third party tools](https://devcenter.bitrise.io/ios/code-signing/#use-a-third-party-tool-to-manage-your-code-signing-files)._
+
+With adding the _fastlane_ step we ensure that you are running on the latest _fastlane_ version, as it is pre-installed on all our VMs. Inside the step you can set the _fastlane_ action and we will run it automatically every time you push a new code change.
```
Comment (janpio) [inline docs/best-practices/continuous-integration/bitrise.md]: How is this related to its headline? Should this be above under the first "Getting started"?

File: docs/best-practices/continuous-integration/bitrise.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+For more configuration options see the `fastlane` step's description in the Workflow Editor!
```
Comment (janpio) [inline docs/best-practices/continuous-integration/bitrise.md]: Can this get context with a link?

File: docs/best-practices/continuous-integration/bitrise.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+### _iOS code signing guide_
+_If you want to use [bitrise.io](https://www.bitrise.io) to store your code signing files, you should just follow the [iOS Code Signing guide here](https://devcenter.bitrise.io/ios/code-signing/)._
```
Comment (janpio) [inline docs/best-practices/continuous-integration/bitrise.md]: Pure fastlane signing (e.g. `match` + `gym`?) does not work?

File: docs/best-practices/continuous-integration/bitrise.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+## How to configure _fastlane match_ for Bitrise
+
+If you want to use [_fastlane match_](https://github.com/fastlane/fastlane/tree/master/match)
```
Comment (janpio) [inline docs/best-practices/continuous-integration/bitrise.md]: Best add `sync_code_signing` here and link to docs at https://docs.fastlane.tools/actions/sync_code_signing/ and https://docs.fastlane.tools/actions/match/

File: docs/best-practices/continuous-integration/bitrise.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+in your [bitrise.io](https://www.bitrise.io/) build you only have to do three things:
+
+1. Make sure that a single SSH key can be used to `git clone` both your main repository (the one
+   you register on [bitrise.io](https://www.bitrise.io/)) and the `match` repository.
+   You can find more info [in this guide](https://github.com/bitrise-io/devcenter/blob/master/faq/adding-projects-with-submodules).
```
Comment (janpio) [inline docs/best-practices/continuous-integration/bitrise.md]: How is this related to submodules? Or does this just accidentally answer the same question?

File: docs/best-practices/continuous-integration/bitrise.md (hunk 2 above)
Comment (RikH81) [reply docs/best-practices/continuous-integration/bitrise.md]: Yup, fixed

File: docs/best-practices/continuous-integration/bitrise.md [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+2. Add an environment variable `MATCH_PASSWORD`, as
+   [described in ](https://github.com/fastlane/fastlane/tree/master/match#encryption-password) [_match_](https://github.com/fastlane/fastlane/tree/master/match#encryption-password) ['s docs](https://github.com/fastlane/fastlane/tree/master/match#encryption-password),
```
Comment (janpio) [inline docs/best-practices/continuous-integration/bitrise.md]: The links here are outdated - this should also link to docs.fastlane.tools (and update your knowledge base as well ;) ). Also no need to have 3 different links.

File: docs/best-practices/continuous-integration/bitrise.md [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+   to specify the `Encryption password` you used for `match`.
+   On [bitrise.io](https://www.bitrise.io/) you should add this as a `Secret Environment Variable`,
```
Comment (janpio) [inline docs/best-practices/continuous-integration/bitrise.md]: less general links to bitrise maybe? Just say "Bitrise" in the text itself.

File: docs/best-practices/continuous-integration/bitrise.md [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+   in the [Workflow Editor](http://devcenter.bitrise.io/docs/add-your-first-step-to-your-apps-workflow).
+   _Make sure to disable_ the `Replace variables in input?` option of the environment
+   variable, to not to cause issues when the value includes the `$` (dollar) sign, which is used
+   for environment variable expansion.
+
+3. Make sure to use `match`'s `readonly` mode, or else `match` will try to connect
+   to the Apple Developer Portal, which requires further authorization (providing additional
+   username and password for Apple Dev Portal login)!
+   * If you use `match` in your `Fastfile` or `fastlane` config: `match(app_identifier: "my.domain", type: "appstore", readonly: true)`
+   * If you use it as a command line tool: `match development --readonly`
```
Comment (janpio) [inline docs/best-practices/continuous-integration/bitrise.md]: Can you just run the CLI in Bitrise?

File: docs/best-practices/continuous-integration/bitrise.md [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+   * More info in `match`'s [official readme / docs](https://docs.fastlane.tools/actions/match/)
+
+## What's next?
```
Comment (janpio) [inline docs/best-practices/continuous-integration/bitrise.md]: This section needs a bit more structure. Paragraph, new lines...

**Comment by janpio** [comment]
Went all in and did a full review of the text.

PS: "Italicization", haha 🙈

**Comment by RikH81** [comment]
> Went all in and did a full review of the text.
> 
> PS: "Italicization", haha 🙈

Awesome, thanks a bunch! Will do a rewrite plus updated our own docs as well just now

File: docs/best-practices/continuous-integration/bitrise.md [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+
+[Bitrise](https://www.bitrise.io) allows you to automatically build, test and deploy your iOS and Android apps.
+
+## How to get started?
+
+Using _fastlane_ for your Bitrise workflow is simple. Just add the [_fastlane_](https://www.bitrise.io/integrations/steps/fastlane) step to your
+workflow [(guide)](https://devcenter.bitrise.io/getting-started/manage-your-bitrise-workflow),
+after the [`Git Clone`](https://www.bitrise.io/integrations/steps/git-clone) step (and any other dependency step).
+
+When you add the _fastlane_ step, we ensure that you are running on the latest _fastlane_ version as it is pre-installed on all our VMs. Inside the step you can set the _fastlane_ action and we will run it automatically every time you push a new code change.
```
Comment (janpio) [inline docs/best-practices/continuous-integration/bitrise.md]:
```suggestion
When you add the _fastlane_ step, Bitrise ensures that you are running on the latest _fastlane_ version as it is pre-installed on all its VMs. Inside the step you can set the _fastlane_ action and Bitrise will run it automatically every time you push a new code change.
```

**Comment by janpio** [comment]
Hey @RikH81, I went through the text once more, _much_ better now. One last paragraph could benefit from getting rid of "we". I resolved all the other questions that were fixed or answered, please take a quick look at what is left now and see if you have answers to those as well.

When my suggestion is accepted and the questions are answered, this and #774 can be merged.

**Comment by rogerluan** [comment]
Closing, as Bitrise is now already present in _fastlane_ docs: https://docs.fastlane.tools/best-practices/continuous-integration/bitrise/
